fili-core/src/test/groovy/com/yahoo/bard/webservice/table/availability/PartitionAvailabilitySpec.groovy

```diff
@@ -0,0 +1,159 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.table.availability
+
+import com.yahoo.bard.webservice.data.config.names.TableName
+import com.yahoo.bard.webservice.table.Column
+import com.yahoo.bard.webservice.table.resolver.DataSourceConstraint
+import com.yahoo.bard.webservice.util.SimplifiedIntervalList
+
+import com.google.common.collect.Sets
+
+import org.joda.time.Interval
+
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import java.util.function.Function
+
+
+/**
+ * Test for partition availability behavior.
+ */
+class PartitionAvailabilitySpec extends Specification{
+
+    PartitionAvailability partitionAvailability
+
+    Availability availability1
+    Availability availability2
+
+    Column column1
+    Column column2
+
+    def setup() {
+        availability1 = Mock(Availability)
+        availability2 = Mock(Availability)
+    }
+
+    @Unroll
+    def "getDataSourceNames returns #expected from #dataSourceNames1 and #dataSourceNames2"() {
+        given:
+        availability1.getDataSourceNames() >> Sets.newHashSet(dataSourceNames1.collect{it -> TableName.of(it)})
+        availability2.getDataSourceNames() >> Sets.newHashSet(dataSourceNames2.collect{it -> TableName.of(it)})
```

Comment:
Can be grooiver: `dataSourceNames.collect {TableName.of(it)} as Set`

- No need to wrap in a HashSet, can just coerce
- No need to declare `it` as a closure param, just use it if it's the only param